sqlite.py

```diff
@@ -3,86 +3,131 @@
     # Note, parameters must be iterable
     cur.execute(select_event_query, (time,))
     return cur.fetchall()
 
 
 def get_events_by_type(conn: sqlite3.Connection, type: str):
     cur = conn.cursor()
-    select_event_query = f'SELECT * FROM {TABLE_NAME} WHERE "event_type"=?'
+    select_event_query = f'SELECT * FROM {DISPENSE_TABLE_NAME} WHERE "event_type"=?'
     # Note, parameters must be iterable
     cur.execute(select_event_query, (type,))
     return cur.fetchall()
 
 
 def get_events_by_type_newer_than(conn: sqlite3.Connection, type: str, time: int):
     cur = conn.cursor()
-    select_event_query = f'SELECT * FROM {TABLE_NAME} WHERE "event_type"=? AND "timestamp" >=?'
+    select_event_query = f'SELECT * FROM {DISPENSE_TABLE_NAME} WHERE "event_type"=? AND "timestamp" >=?'
     # Note, parameters must be iterable
     cur.execute(select_event_query, (type, time))
     return cur.fetchall()
 
 
 def get_events_by_type_in_range(conn: sqlite3.Connection, type: str, lower_time: int, upper_time: int):
     cur = conn.cursor()
-    select_event_query = f'SELECT * FROM {TABLE_NAME} WHERE "event_type"=? AND "timestamp" >=? AND "timestamp" <=?'
+    select_event_query = f'SELECT * FROM {DISPENSE_TABLE_NAME}' \
+                         f' WHERE "event_type"=? AND "timestamp" >=? AND "timestamp" <=?'
     # Note, parameters must be iterable
     cur.execute(select_event_query, (type, lower_time, upper_time))
     return cur.fetchall()
 
 
 def get_events_by_item(conn: sqlite3.Connection, item: str):
     cur = conn.cursor()
-    select_event_query = f'SELECT * FROM {TABLE_NAME} WHERE "event_item"=?'
+    select_event_query = f'SELECT * FROM {DISPENSE_TABLE_NAME} WHERE "event_item"=?'
     # Note, parameters must be iterable
     cur.execute(select_event_query, (item,))
     return cur.fetchall()
 
 
 def get_events(conn: sqlite3.Connection):
     cur = conn.cursor()
-    select_query = f'SELECT * FROM {TABLE_NAME}'
+    select_query = f'SELECT * FROM {DISPENSE_TABLE_NAME}'
     cur.execute(select_query)
     return cur.fetchall()
 
 
 def get_last_event(conn: sqlite3.Connection):
     cur = conn.cursor()
-    select_last_query = f'SELECT * FROM    {TABLE_NAME} WHERE   timestamp = (SELECT MAX(timestamp)  FROM {TABLE_NAME});'
+    select_last_query = f'SELECT * FROM    {DISPENSE_TABLE_NAME} ' \
+                        f'WHERE   timestamp = (SELECT MAX(timestamp)  FROM {DISPENSE_TABLE_NAME});'
+    cur.execute(select_last_query)
+    return cur.fetchall()
+
+
+def get_last_event_by_type(conn: sqlite3.Connection, type: str):
+    cur = conn.cursor()
+    select_last_query_by_type = f'SELECT * FROM {DISPENSE_TABLE_NAME} ' \
+                                f'WHERE "event_type"=? ORDER BY timestamp DESC LIMIT 1'
+    cur.execute(select_last_query_by_type, (type,))
+    return cur.fetchall()
+
+
+def get_last_event_by_type_older_than(conn: sqlite3.Connection, type: str, upper_time: int):
```

Comment:
Should have an inverse function as well.

Reply:
Function added in commit fde9d383289f262cd6dd454e4792851f4a3b0306

Reply:
Added in ffc1ae70725ea53dd132dfccbc369c1fc71ba2df